### DISCUSSION TITLE
`` vs '' vs ‘’

### DISCUSSION THREAD

**Discussion by danieldaeschle** [vlang/v#7242] [Ideas]
We are using ``, typescript uses '' and linux (https://unix.stackexchange.com/questions/126501/cp-what-quotation-marks-are-used-in-error-messages/126511) uses ‘’ for errors.

I like the last ones more than ours. It just looks better... What about changing it?

**Comment by spytheman** [comment]
The third kind of quotes: ‘’ are not easy to type on most keyboards.
'' is already used for most strings, so using them inside the strings themselves, will require escaping.
In effect these constraints pretty much leave only the back ticks as an easy to use character for that specific purpose.

**Reply by danieldaeschle** [reply]
That's true. We could replace them on errors with any other variant.

I would also suggest UI/UX over programming easieness. Having a nice interface to the user gives a feeling of quality.

**Reply by spytheman** [reply]
For me, a programmer tool, that produces easily greppable errors is much superior to another, that sacrifices that feature, for the sake of some idealized user friendliness. NB: I am a user too.

**Reply by danieldaeschle** [reply]
That's right. But we want to reach users that are used to use javascript as well. not only low levels devs. It's just a perspective from me, a UI programmer.

**Reply by spytheman** [reply]
hm, I do not see the connection between the types of used quotes in errors, and the programmer background ...

**Comment by Delta456** [comment]
` ‘’ ` isn't ASCII at all, many programmers use ASCII based keyboards plus we only allow ASCII names and as we are also supporting older systems, some will not have Unicode for console output like Windows and Mac. I already like the use of backticks for treating them as `byte` and `rune`. This is pretty amazing like how Go uses backticks for denoting raw strings.

Thus, backticks are a win!
